staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -102,11 +102,13 @@ func (e *eventBroadcasterImpl) finishSeries() {
 	for isomorphicKey, event := range e.eventCache {
 		eventSerie := event.Series
 		if eventSerie != nil {
-			if eventSerie.LastObservedTime.Time.Add(finishTime).Before(time.Now()) {
+			if eventSerie.LastObservedTime.Time.Before(time.Now().Add(-finishTime)) {
```

Comment:
I realized now that I don't really understand why we need to recordEvent here - nothing should change since the previous one, right?

Reply:
This avoids users to wait (at worst) 30min to see the final event count, I think we should keep it

Reply:
Hmm - I think I don't fully understand it.
1. Yes - i agree that we may update the count or sth here.
2. But if we wouldn't do that here, it would never happen, right? i.e. - I don't understand the 30m part of your comment.

[But no matter what, please add a comment in the code why we're doing this]

Reply:
There are 2 things at play

1 - refreshing TTL on events which happens every 30min
2 - Finishing series to write the final count that happens each 6min

if we drop 2 we'd only have a refresh each 30min which causes too much delay. I'll add a comment to explain this.